snuba/query/formatters/anonymize.py

```diff
@@ -0,0 +1,270 @@
+
+    def _visit_simple_source(self, data_source: SimpleDataSource) -> StringNode:
+        sample_val = getattr(
+            data_source, "sample", getattr(data_source, "sampling_rate", None)
+        )
+        sample_str = f" SAMPLE {sample_val}" if sample_val is not None else ""
+        return StringNode(f"{data_source.human_readable_id}{sample_str}")
+
+    def _visit_join(self, data_source: JoinClause[Entity]) -> StringNode:
+        return self.visit_join_clause(data_source)
+
+    def _visit_simple_query(
+        self, data_source: ProcessableQuery[Entity]
+    ) -> FormattedSubQuery:
+        assert isinstance(data_source, LogicalQuery)
+        return FormattedSubQuery(
+            _format_query_content(data_source, self.__expression_formatter_type)
+        )
+
+    def _visit_composite_query(
+        self, data_source: CompositeQuery[Entity]
+    ) -> FormattedSubQuery:
+        return FormattedSubQuery(
+            _format_query_content(data_source, self.__expression_formatter_type)
+        )
+
+    def visit_individual_node(self, node: IndividualNode[Entity]) -> StringNode:
+        return StringNode(f"{node.alias}, {self.visit(node.data_source)}")
+
+    def visit_join_clause(self, node: JoinClause[Entity]) -> StringNode:
+        left = f"LEFT {node.left_node.accept(self)}"
+        type = f"TYPE {node.join_type}"
+        right = f"RIGHT {node.right_node.accept(self)}"
+        on = "".join(
+            [
+                f"{c.left.table_alias}.{c.left.column} {c.right.table_alias}.{c.right.column}"
+                for c in node.keys
+            ]
+        )
+
+        return StringNode(f"{left} {type} {right} ON {on}")
+
+
+def _format_query_content(
+    query: Union[LogicalQuery, CompositeQuery[Entity]],
+    expression_formatter_type: Type[ExpressionVisitor[str]],
+) -> Sequence[FormattedNode]:
+    formatter = expression_formatter_type()
+
+    return [
+        v
+        for v in [
+            _format_select(query, formatter),
+            _format_groupby(query, formatter),
+            _format_orderby(query, formatter),
+            _build_optional_string_node("ARRAY JOIN", query.get_arrayjoin(), formatter),
+            _build_optional_string_node("WHERE", query.get_condition(), formatter),
+            _build_optional_string_node("HAVING", query.get_having(), formatter),
+            _format_limitby(query, formatter),
+            _format_limit(query, formatter),
+            _format_offset(query, formatter),
+            PaddingNode(
+                "FROM",
+                StringQueryFormatter(expression_formatter_type).visit(
+                    query.get_from_clause()
+                ),
+            ),
+        ]
+        if v is not None
+    ]
+
+
+def _format_select(
+    query: AbstractQuery, formatter: ExpressionVisitor[str]
+) -> StringNode:
+    selected_cols = [
+        e.expression.accept(formatter) for e in query.get_selected_columns()
+    ]
+    print(type(selected_cols[0]))
```

Comment:
💀

Reply:
There are some small changes that I couldn't find a good way to rectify -- going to attempt that now